binding-mri/binding-util.h

```diff
@@ -243,11 +243,12 @@ rb_int_arg(VALUE arg, int *out, int argPos = 0)
 		break;
 
 	case RUBY_T_FIXNUM :
+	case RUBY_T_RATIONAL :
```

Comment:
Have you tested this? `FIX2INT()` only works with fixnums.

Reply:
Yes,
the only games that I encountered this problem are some rgss1 games,
that use a plugin to determine the characters position.
float value (X,Y position of the character) & rational value (expected X,Y new value)

Reply:
Yeah, RGSS3 also supports rational numbers but your change may only appear to work. Could you please try something like `Rect.new(Rational(10, 10), 0, 0, 0)` and check the value of its `x` attribute? I can't compile mkxp at the moment and there's no test suite or build bot.

IMHO the type check isn't necessary anyway. `*out = NUM2INT(arg);` is good enough (and `rb_rescue()` if the custom error message should be kept). This would add support for complex numbers too: `Rect.new(Complex(10, 0), 0, 0, 0)`.

Reply:
Can you give another test?
My Falcon-mkxp has unlocked resolution, so the x result will always deffer, depending on the game resolution & monitor resolution & position of the window.

Reply:
X should always be 1. But it can be tested with any other integer attribute:
```ruby
s = Sprite.new
s.z = Rational(10, 2)
p s.z # => 5
```

Reply:
Thank you.
I created 2 events to call the test scripts and both of them
return positive random(?) values with 8 digits(!) instead of returning x=1 & s=5.